src/convert.rs

```diff
@@ -0,0 +1,192 @@
+use num::Float;
+
+use {Alpha, Rgb, Luma, Xyz, Yxy, Lab, Lch, Hsv, Hsl, Color};
+
+///Provides conversion between the colors.
+///Color Conver requires from and into xyz and derives others from this.
+///
+///This is not optimal in all cases i.e Hsl -> Rgb using this method will
+///result in Hsl -> Rgb -> Xyz -> Rgb. To avoid this the Rgb derivatives
+///like Hsl and Hsv, convert through Rgb as default and Lab derivatives
+///like Lch convert through Lab as defined in the code below.
+pub trait ColorConvert<T>: Sized
+    where T: Float,
+{
+    //To implement ColorConvert for a type, a few things must by kept in mind.
+    //
+    //First, derived colors like Hsl must override into_rgb and from_rgb and
+    //convert into_xyz and from_xyz via Rgb, as discussed above.
+    //
+    //Second, the from and into for the same color must override the default.
+    //For example, from_rgb and into_rgb for Rgb will convert via Xyz which
+    //needs to be overridden with *input_color and *self to avoid the unnecessary
+    //converison.
+    //
+    //Third, if a direct transform exists between the colors, then it should be
+    //overridden in both the colors. For example, Luma has direct conversion to
+    //Rgb. So into_rgb and from_rgb conversion is implemented in Luma and into_luma
+    //and from_luma is overridden in Rgb to use the conversion from Luma.
+
+    ///Convert from XYZ color space
+    fn from_xyz(&Xyz<T>) -> Self;
+
+    ///Convert into xyz space
```

Comment:
The capitalization of XYZ this comment and the one above is inconsistent with the rest. The conventions I try to follow is to write `Xyz`, `Lab`, etc. if I'm talking about the Rust types, and write XYZ, L*a*b*, etc. if I'm talking about the color spaces in general.